arroyo-worker/src/operators/windows.rs

```diff
@@ -425,6 +425,14 @@ impl<K: Key, T: Data, OutT: Data> SessionWindowFunc<K, T, OutT> {
             .clear_time_range(&mut key, window.start, window.end)
             .await;
     }
+
+    async fn handle_checkpoint(
```

Comment:
Under what conditions does an operator need to define `handle_checkpoint` itself? Should we remove the empty one in the macro?

Reply:
If it is using table types that buffer data, it needs to flush those. In the state store changes I have planned there's a change that pushes this back into the state store itself, which will be nice.